src/bridge/cockpitchannel.c

```diff
@@ -112,7 +112,9 @@ G_DEFINE_TYPE (CockpitChannel, cockpit_channel, G_TYPE_OBJECT);
 static gboolean
 on_idle_prepare (gpointer data)
 {
-  cockpit_channel_prepare (data);
+  CockpitChannel *self = COCKPIT_CHANNEL (data);
+  cockpit_channel_prepare (self);
+  g_object_unref (self);
```

Comment:
If we are trying to have a guard across the prepare call we should do that right here and not split it across the idle call. The prepare_tag is already guarded by a g_source_remove().